### PR TITLE
Allow to set SL_API_ENDPOINT environment variable

### PR DESCRIPTION
**Pull request by mattcui**
To address [issue 118](https://github.com/maximilien/softlayer-go/issues/118), users can set environment variable SL_API_ENDPOINT to change the softlayer endpoint server.

### CODE REVIEW

**Comment by maximilien** [comment]
OK @mattcui but what about a simple test that verifies that when `SL_API_ENDPOINT` is set correct value is used in endpoint and when not default. Feel fere to add a `GetApiEndpoint() string` to the `client.go` `Client` interface and to the implementation in `softlayer_client.go` so that way we can test it.

Thanks,

Max

**Comment by mattcui** [comment]
@maximilien It's not easy to add it into Client interface, since the api endpoint url needs to be set in [NewSoftLayerClient](https://github.com/maximilien/softlayer-go/blob/master/client/softlayer_client.go#L24), I made some enhancement there, changed it to a public function, there will verify if it's a known api endpoint server, and set to "api.softlayer.com" if it's not set or it's set wrongly. I also added 2 unit test cases. Please review. Thanks.

File: client/softlayer_client.go [hunk 1]
```diff
@@ -174,6 +176,24 @@ func (slc *SoftLayerClient) GetSoftLayer_Dns_Domain_ResourceRecord_Service() (so
 
 //Private methods
```
Comment (maximilien) [inline client/softlayer_client.go]: That's not private... capital letter `GetSLApiEndpoint()`

File: client/softlayer_client_test.go [hunk 2]
```diff
@@ -187,4 +187,20 @@ var _ = Describe("SoftLayerClient", func() {
 			Expect(hardwareService).ToNot(BeNil())
 		})
 	})
+
+	Context("#getApiEndpoint, when SL_API_ENDPOINT is set correctly", func() {
```
Comment (maximilien) [inline client/softlayer_client_test.go]: This should be: `"#GetApiEndpoint, when SL_API_ENDPOINT is set correctly"`

**Comment by maximilien** [comment]
@mattcui Merging and I'll address my two comments. Thx

**Pull request merged**